Mixpanel/MPABTestDesignerConnection.m

```diff
@@ -218,21 +222,24 @@ - (void)webSocket:(MPWebSocket *)webSocket didCloseWithCode:(NSInteger)code reas
     _open = NO;
     if (_connected) {
         _connected = NO;
-        [self reconnect:YES];
+        [self reconnect:YES maxInterval:10 maxRetries:10];
         if (_disconnectCallback) {
             _disconnectCallback();
         }
+    } else if (_keepTrying) {
+        [self reconnect:YES maxInterval:30 maxRetries:40];
     }
 }
 
-- (void)reconnect:(BOOL)initiate
+- (void)reconnect:(BOOL)initiate maxInterval:(int)maxInterval maxRetries:(int)maxRetries
 {
     static int retries = 0;
-    if (self.sessionEnded || _connected || retries >= 10) {
+    
+    if (self.sessionEnded || _connected || retries >= maxRetries) {
         // If we deliberately closed the connection, or are already connected
         // or we tried too many times, then stop retrying.
         retries = 0;
-    } else if(initiate ^ (retries > 0)) {
+    } else if (initiate ^ (retries > 0)) {
```

Comment:
Actually the 4th case, where both are NO is also kindof important. It means we successfully reconnected, or the max retries was reached, and we want to stop the reconnect loop.